lib/types/inet-address.js

```diff
@@ -37,42 +37,55 @@ InetAddress.fromString = function (value) {
   if (!value) {
     return new InetAddress(new Buffer([0, 0, 0, 0]));
   }
-  var pattern = /^\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}$|^[\da-f:]+$/i;
-  if (!pattern.test(value)) {
-    throw new TypeError('Value could not be parsed as InetAddress: ' + value);
-  }
-  var parts = value.split('.');
-  if (parts.length === 4) {
-    //IPv4
+  var ipv4Pattern = /^\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}$/;
+  var ipv6Pattern = /^[\da-f:.]+$/i;
```

Comment:
This regex will not match v6 addresses with %interface postfix.
Also, and this may be by intent given the 4-in-6 tests below, but this pattern will match v4 addresses.